### PR TITLE
ROX-25990: Add debug logging to flaky test

### PR DESCRIPTION
**Pull request by Maddosaurus**
### Description

To further debug the flaky `Test_SensorIntermediateRuntimeEvents`, debug logs are added to provide insight into the state of Network Flows at test time.

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [x] modified existing tests

#### How I validated my change
As this is as work in progress to collect additional information, no validation has been conducted at this point in time.

### CODE REVIEW

**Review by vikin91 — APPROVED** [review]

**Review by jschnath — APPROVED** [review]

**Comment by vikin91** [comment]
The CI failed because of exactly that flake ;) https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/stackrox_stackrox/12781/pull-ci-stackrox-stackrox-master-gke-sensor-integration-tests/1838223071769530368

**Comment by vikin91** [comment]
Previous failure due to: `panic: informer has already started`

/test gke-sensor-integration-tests

**Comment by vikin91** [comment]
Example output of passing test: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/stackrox_stackrox/12781/pull-ci-stackrox-stackrox-master-gke-sensor-integration-tests/1841038050541441024/artifacts/gke-sensor-integration-tests/stackrox-stackrox-e2e-test/build-log.txt

**Comment by vikin91** [comment]
Failures triggered by the flake: 
- https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/stackrox_stackrox/12781/pull-ci-stackrox-stackrox-master-gke-sensor-integration-tests/1841719595031859200
-

File: sensor/common/networkflow/manager/manager_impl.go [hunk 1]
```diff
@@ -469,15 +472,18 @@ func (m *networkFlowManager) enrichAndSend() {
...
 			NetworkFlowUpdate: protoToSend,
 		},
 	}) {
+		log.Info("Flow has been enqueued in sensorUpdates")
 		m.updateConnectionStates(currentConns, currentEndpoints)
 		metrics.IncrementTotalNetworkFlowsSentCounter(len(protoToSend.Updated))
 		metrics.IncrementTotalNetworkEndpointsSentCounter(len(protoToSend.UpdatedEndpoints))
+	} else {
+		log.Info("Flow has NOT been enqueued in sensorUpdates")
 	}
```
Comment (vikin91) [inline sensor/common/networkflow/manager/manager_impl.go:487]: None of these lines are visible in the log of a failing run: https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/stackrox_stackrox/12781/pull-ci-stackrox-stackrox-master-gke-sensor-integration-tests/1841719595031859200

**Comment by vikin91** [comment]
Fragment of logs where the tests pass:

```
    runtime_test.go:156: Expecting Network Flow for 6e82d8c3-d648-4334-9943-a74a16f78d83 -> c0656e2d-a119-44f4-963b-64321905ebb0
    runtime_test.go:159: Processing Flow: type:DEPLOYMENT  id:"6e82d8c3-d648-4334-9943-a74a16f78d83" - type:DEPLOYMENT  id:"c0656e2d-a119-44f4-963b-64321905ebb0"
pkg/providers: 2024/10/01 09:36:12.859274 metadata.go:40: Error: getting cloud provider metadata error: AWS: retrieving AWS EC2 metadata errors: [retrieving RSA-2048 signature: operation error ec2imds: GetDynamicData, exceeded maximum number of attempts, 3, request send failed, Get "http://169.254.169.254/latest/dynamic/instance-identity/rsa2048": dial tcp 169.254.169.254:80: connect: connection refused, operation error ec2imds: GetInstanceIdentityDocument, exceeded maximum number of attempts, 3, request send failed, Get "http://169.254.169.254/latest/dynamic/instance-identity/document": dial tcp 169.254.169.254:80: connect: connection refused]
kubernetes/clusterstatus: 2024/10/01 09:36:12.859371 updater.go:315: Info: No cloud provider metadata was found
pkg/cloudproviders/aws: 2024/10/01 09:36:13.916326 metadata.go:55: Warn: Could not verify AWS public certificate: retrieving RSA-2048 signature: operation error ec2imds: GetDynamicData, exceeded maximum number of attempts, 3, request send failed, Get "http://169.254.169.254/latest/dynamic/instance-identity/rsa2048": dial tcp 169.254.169.254:80: connect: connection refused
common/networkflow/manager: 2024/10/01 09:36:15.736843 manager_impl.go:427: Info: enriching connections tick
common/networkflow/manager: 2024/10/01 09:36:15.737537 manager_impl.go:475: Info: Sending flow update to central: updated:{props:{src_entity:{type:DEPLOYMENT  id:"6e82d8c3-d648-4334-9943-a74a16f78d83"}  dst_entity:{type:DEPLOYMENT  id:"c0656e2d-a119-44f4-963b-64321905ebb0"}  dst_port:80  l4protocol:L4_PROTOCOL_TCP}}  time:{seconds:1727775375  nanos:737089517}
common/networkflow/manager: 2024/10/01 09:36:15.737593 manager_impl.go:481: Info: Flow has been enqueued in sensorUpdates
pkg/providers: 2024/10/01 09:36:15.919983 metadata.go:40: Error: getting cloud provider metadata error: AWS: retrieving AWS EC2 metadata errors: [retrieving RSA-2048 signature: operation error ec2imds: GetDynamicData, exceeded maximum number of attempts, 3, request send failed, Get "http://169.254.169.254/latest/dynamic/instance-identity/rsa2048": dial tcp 169.254.169.254:80: connect: connection refused, operation error ec2imds: GetInstanceIdentityDocument, exceeded maximum number of attempts, 3, request send failed, Get "http://169.254.169.254/latest/dynamic/instance-identity/document": dial tcp 169.254.169.254:80: connect: connection refused]
kubernetes/clusterstatus: 2024/10/01 09:36:15.920100 updater.go:315: Info: No cloud provider metadata was found
--- PASS: Test_SensorIntermediateRuntimeEvents (130.41s)
```

Fragment of logs where the test fails:

```
 runtime_test.go:156: Expecting Network Flow for 6657b4fe-bab4-4aab-b50e-56e0543859e1 -> ead405f6-061e-4cb3-b729-04e67637eb4d
pkg/providers: 2024/10/03 06:17:59.959505 metadata.go:40: Error: getting cloud provider metadata error: AWS: retrieving AWS EC2 metadata errors: [retrieving RSA-2048 signature: operation error ec2imds: GetDynamicData, exceeded maximum number of attempts, 3, request send failed, Get "http://169.254.169.254/latest/dynamic/instance-identity/rsa2048": dial tcp 169.254.169.254:80: connect: connection refused, operation error ec2imds: GetInstanceIdentityDocument, exceeded maximum number of attempts, 3, request send failed, Get "http://169.254.169.254/latest/dynamic/instance-identity/document": dial tcp 169.254.169.254:80: connect: connection refused]
kubernetes/clusterstatus: 2024/10/03 06:17:59.959636 updater.go:315: Info: No cloud provider metadata was found
pkg/cloudproviders/aws: 2024/10/03 06:18:00.733779 metadata.go:55: Warn: Could not verify AWS public certificate: retrieving RSA-2048 signature: operation error ec2imds: GetDynamicData, exceeded maximum number of attempts, 3, request send failed, Get "http://169.254.169.254/latest/dynamic/instance-identity/rsa2048": dial tcp 169.254.169.254:80: connect: connection refused
pkg/providers: 2024/10/03 06:18:02.737145 metadata.go:40: Error: getting cloud provider metadata error: AWS: retrieving AWS EC2 metadata errors: [retrieving RSA-2048 signature: operation error ec2imds: GetDynamicData, exceeded maximum number of attempts, 3, request send failed, Get "http://169.254.169.254/latest/dynamic/instance-identity/rsa2048": dial tcp 169.254.169.254:80: connect: connection refused, operation error ec2imds: GetInstanceIdentityDocument, exceeded maximum number of attempts, 3, request send failed, Get "http://169.254.169.254/latest/dynamic/instance-identity/document": dial tcp 169.254.169.254:80: connect: connection refused]
kubernetes/clusterstatus: 2024/10/03 06:18:02.737249 updater.go:315: Info: No cloud provider metadata was found
common/networkflow/manager: 2024/10/03 06:18:03.408385 manager_impl.go:427: Info: enriching connections tick
kubernetes/clusterhealth: 2024/10/03 06:18:26.738147 updater.go:171: Error: Errors while getting collector info: [unable to find collector DaemonSet in namespace "stackrox": daemonsets.apps "collector" not found]
kubernetes/clusterhealth: 2024/10/03 06:18:26.794448 updater.go:194: Error: Errors while getting admission control info: [unable to find admission control deployments in namespace "stackrox": unable to find admission control deployments in namespace "stackrox": deployments.apps "admission-control" not found]
common/networkflow/manager: 2024/10/03 06:18:33.409189 manager_impl.go:427: Info: enriching connections tick
kubernetes/clusterhealth: 2024/10/03 06:18:56.676088 updater.go:171: Error: Errors while getting collector info: [unable to find collector DaemonSet in namespace "stackrox": daemonsets.apps "collector" not found]
kubernetes/clusterhealth: 2024/10/03 06:18:56.709727 updater.go:194: Error: Errors while getting admission control info: [unable to find admission control deployments in namespace "stackrox": unable to find admission control deployments in namespace "stackrox": deployments.apps "admission-control" not found]
    runtime_test.go:166: 
        	Error Trace:	/go/src/github.com/stackrox/stackrox/sensor/tests/connection/runtime/runtime_test.go:166
        	            				/go/src/github.com/stackrox/stackrox/sensor/tests/helper/helper.go:384
        	            				/go/src/github.com/stackrox/stackrox/sensor/tests/helper/helper.go:337
        	            				/go/src/github.com/stackrox/stackrox/sensor/tests/helper/helper.go:230
        	            				/go/src/github.com/stackrox/stackrox/sensor/tests/connection/runtime/runtime_test.go:64
        	Error:      	Received unexpected error:
        	            	timeout (1m0s) reached waiting for message
        	            	github.com/stackrox/rox/sensor/tests/helper.(*TestContext).WaitForMessageWithMatcher
        	            		/go/src/github.com/stackrox/stackrox/sensor/tests/helper/helper.go:546
        	            	github.com/stackrox/rox/sensor/tests/connection/runtime.Test_SensorIntermediateRuntimeEvents.func1
        	            		/go/src/github.com/stackrox/stackrox/sensor/tests/connection/runtime/runtime_test.go:157
        	            	github.com/stackrox/rox/sensor/tests/helper.(*TestContext).runBare
        	            		/go/src/github.com/stackrox/stackrox/sensor/tests/helper/helper.go:384
        	            	github.com/stackrox/rox/sensor/tests/helper.(*TestContext).run
        	            		/go/src/github.com/stackrox/stackrox/sensor/tests/helper/helper.go:337
        	            	github.com/stackrox/rox/sensor/tests/helper.(*TestContext).RunTest
        	            		/go/src/github.com/stackrox/stackrox/sensor/tests/helper/helper.go:230
        	            	github.com/stackrox/rox/sensor/tests/connection/runtime.Test_SensorIntermediateRuntimeEvents
        	            		/go/src/github.com/stackrox/stackrox/sensor/tests/connection/runtime/runtime_test.go:64
        	            	testing.tRunner
        	            		/usr/local/go/src/testing/testing.go:1689
        	            	runtime.goexit
        	            		/usr/local/go/src/runtime/asm_amd64.s:1695
        	Test:       	Test_SensorIntermediateRuntimeEvents
    runtime_test.go:167: 
        	Error Trace:	/go/src/github.com/stackrox/stackrox/sensor/tests/connection/runtime/runtime_test.go:167
        	            				/go/src/github.com/stackrox/stackrox/sensor/tests/helper/helper.go:384
        	            				/go/src/github.com/stackrox/stackrox/sensor/tests/helper/helper.go:337
        	            				/go/src/github.com/stackrox/stackrox/sensor/tests/helper/helper.go:230
        	            				/go/src/github.com/stackrox/stackrox/sensor/tests/connection/runtime/runtime_test.go:64
        	Error:      	Expected value not to be nil.
        	Test:       	Test_SensorIntermediateRuntimeEvents
    helper.go:786: timeout reached waiting for violation state (test-flow): expected set ([test-flow]) differs from actual ([test-pi-curl])
common/networkflow/manager: 2024/10/03 06:19:03.408732 manager_impl.go:427: Info: enriching connections tick
--- FAIL: Test_SensorIntermediateRuntimeEvents (190.79s)
```

**Comment by vikin91** [comment]
I have analyzed the logs of failing and passing tests and the current set of messages does not produce any hints that we could explore. What we do know is that the issue is not caused around the queue `sensorUpdates` as the runs of failing tests never get to that point (i.e., we never see `Sending flow update to central:`).

**Comment by vikin91** [comment]
Finally it flaked 🎉 https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/stackrox_stackrox/12781/pull-ci-stackrox-stackrox-master-gke-sensor-integration-tests/1844314284050354176

**Comment by vikin91** [comment]
Flaked in two different ways: 

1. https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/stackrox_stackrox/12781/pull-ci-stackrox-stackrox-master-gke-sensor-integration-tests/1844471153129689088
2. https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/stackrox_stackrox/12781/pull-ci-stackrox-stackrox-master-gke-sensor-integration-tests/1844494158673219584

**Comment by vikin91** [comment]
closing in favor of https://github.com/stackrox/stackrox/pull/12989

**Pull request closed without merging**